README.md

```diff
@@ -103,8 +102,294 @@ $variables = [
+ *              ]),
+ *          ]),
+ *      ]),
+ *  ]);
+ */
+```
+
+Now we can modify the mutation data using the following methods:
+* add(): Adds an Item to a Collection.
+* set(): Updates some values of an Item. It also works on Collections, updating all its Items.
+* filter(): Filters the Items of a Collection.
+
+``` php
+$mutation->book->chapters->upsert->filter(['id_chapter' => 2])->pages->upsert->add([
+    'id_book'           => 'f7cfd732-e3d8-3642-a919-ace8c38c2c6d',
+    'id_chapter'        => 2,
+    'id_page'           => 3,
+    'has_illustrations' => false,
+]);
+$mutation->book->chapters->upsert->pages->upsert->filter([
+    'id_chapter' => 2,
+    'id_page'    => 2,
+])->set(['has_illustrations' => true]);
+
+/**
+ * $mutation = [
+ *     'book' => [
+ *         'id_book'   => 'f7cfd732-e3d8-3642-a919-ace8c38c2c6d',
+ *         'id_author' => 1234,
+ *         'genre'     => 'adventure',
+ *         'chapters'  => [
+ *             'upsert' => [
+ *                 [
+ *                     'id_book'    => 'f7cfd732-e3d8-3642-a919-ace8c38c2c6d',
+ *                     'id_chapter' => 1,
+ *                     'name'       => 'Chapter One',
+ *                     'pov'        => 'first person',
+ *                     'pages'      => new QueryCollection([]),
+ *                 ],
+ *                 [
+ *                     'id_book'    => 'f7cfd732-e3d8-3642-a919-ace8c38c2c6d',
+ *                     'id_chapter' => 2,
+ *                     'name'       => 'Chapter two',
+ *                     'pov'        => 'third person',
+ *                     'pages'      => [
+ *                         'upsert' => [
+ *                             [
+ *                                 'id_book'           => 'f7cfd732-e3d8-3642-a919-ace8c38c2c6d',
+ *                                 'id_chapter'        => 2,
+ *                                 'id_page'           => 1,
+ *                                 'has_illustrations' => false,
+ *                             ],
+ *                             [
+ *                                 'id_book'           => 'f7cfd732-e3d8-3642-a919-ace8c38c2c6d',
+ *                                 'id_chapter'        => 2,
+ *                                 'id_page'           => 2,
+ *                                 'has_illustrations' => true,
+ *                             ],
+ *                             [
+ *                                 'id_book'           => 'f7cfd732-e3d8-3642-a919-ace8c38c2c6d',
+ *                                 'id_chapter'        => 2,
+ *                                 'id_page'           => 3,
+ *                                 'has_illustrations' => false,
+ *                             ],
+ *                         ],
+ *                     ],
+ *                 ],
+ *             ],
+ *         ],
+ *     ],
+ * ];
+ */
+```
+
+Finally, the modified mutation data can be passed to the GraphQL client to execute the mutation.
```

Comment:
Maybe is worth to say that only the modified items (and upper hierarchy) is returned as mutation variable.

Reply:
Moving the part you say to this point of the doc.